### ISSUE TITLE
Can't run training

### ISSUE THREAD

**Issue by hudvin** [facebookresearch/deepmask#18]
I'm trying to follow Training Your Own Model paragraph and get:

[kontiki@sanctum deepmask (master ✗)]$ th train.lua -datadir data                                                                                                                                                                  (master|…)
-- ignore option rundir	
-- ignore option dm	
-- ignore option reload	
-- ignore option gpu	
-- ignore option datadir	
| running in directory /media/3tb/data/downloads/nix/lab/deepmask/exps/deepmask/exp	
| number of paramaters trunk: 15198016	
| number of paramaters mask branch: 1608768	
| number of paramaters score branch: 526337	
| number of paramaters total: 17333121	
/opt/torch/install/bin/luajit: /opt/torch/install/share/lua/5.1/threads/threads.lua:183: [thread 2 callback] /opt/torch/install/share/lua/5.1/coco/CocoApi.lua:126: assertion failed!
stack traceback:
	[C]: in function 'assert'
	/opt/torch/install/share/lua/5.1/coco/CocoApi.lua:126: in function '__init'
	/opt/torch/install/share/lua/5.1/torch/init.lua:91: in function </opt/torch/install/share/lua/5.1/torch/init.lua:87>
	[C]: in function 'CocoApi'
	/media/3tb/data/downloads/nix/lab/deepmask/DataSampler.lua:25: in function '__init'
	/opt/torch/install/share/lua/5.1/torch/init.lua:91: in function </opt/torch/install/share/lua/5.1/torch/init.lua:87>
	[C]: in function 'DataSampler'
	/media/3tb/data/downloads/nix/lab/deepmask/DataLoader.lua:36: in function </media/3tb/data/downloads/nix/lab/deepmask/DataLoader.lua:30>
	[C]: in function 'xpcall'
	/opt/torch/install/share/lua/5.1/threads/threads.lua:234: in function 'callback'
	/opt/torch/install/share/lua/5.1/threads/queue.lua:65: in function </opt/torch/install/share/lua/5.1/threads/queue.lua:41>
	[C]: in function 'pcall'
	/opt/torch/install/share/lua/5.1/threads/queue.lua:40: in function 'dojob'
	[string "  local Queue = require 'threads.queue'..."]:13: in main chunk
stack traceback:
	[C]: in function 'error'
	/opt/torch/install/share/lua/5.1/threads/threads.lua:183: in function 'dojob'
	/opt/torch/install/share/lua/5.1/threads/threads.lua:264: in function 'synchronize'
	/opt/torch/install/share/lua/5.1/threads/threads.lua:142: in function 'specific'
	/opt/torch/install/share/lua/5.1/threads/threads.lua:125: in function 'Threads'
	/media/3tb/data/downloads/nix/lab/deepmask/DataLoader.lua:40: in function '__init'
	/opt/torch/install/share/lua/5.1/torch/init.lua:91: in function </opt/torch/install/share/lua/5.1/torch/init.lua:87>
	[C]: in function 'DataLoader'
	/media/3tb/data/downloads/nix/lab/deepmask/DataLoader.lua:21: in function 'create'
	train.lua:101: in main chunk
	[C]: in function 'dofile'
	/opt/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:145: in main chunk
	[C]: at 0x00406670

**Comment by tylin** [comment]
From https://github.com/pdollar/coco/blob/master/LuaAPI/CocoApi.lua#L126
It looks like the path to annotations file is incorrect or does not exist.

**Comment by hudvin** [comment]
Oh, I haven't unpacked datasets. But now I have another one error:

[kontiki@sanctum deepmask (master ✗)]$ th train.lua                  (master|…)
-- ignore option rundir	
-- ignore option dm	
-- ignore option reload	
-- ignore option gpu	
-- ignore option datadir	
| running in directory /media/3tb/data/downloads/nix/lab/deepmask/exps/deepmask/exp	
| number of paramaters trunk: 15198016	
| number of paramaters mask branch: 1608768	
| number of paramaters score branch: 526337	
| number of paramaters total: 17333121	
convert: data//annotations/instances_train2014.json --> .t7 [please be patient]
convert: data//annotations/instances_train2014.json --> .t7 [please be patient]
FATAL THREAD PANIC: (pcall) not enough memory
/opt/torch/install/bin/luajit: /opt/torch/install/share/lua/5.1/threads/threads.lua:183: [thread 2 callback] not enough memory
stack traceback:
	[C]: in function 'error'
	/opt/torch/install/share/lua/5.1/threads/threads.lua:183: in function 'dojob'
	/opt/torch/install/share/lua/5.1/threads/threads.lua:264: in function 'synchronize'
	/opt/torch/install/share/lua/5.1/threads/threads.lua:142: in function 'specific'
	/opt/torch/install/share/lua/5.1/threads/threads.lua:125: in function 'Threads'
	/media/3tb/data/downloads/nix/lab/deepmask/DataLoader.lua:40: in function '__init'
	/opt/torch/install/share/lua/5.1/torch/init.lua:91: in function </opt/torch/install/share/lua/5.1/torch/init.lua:87>
	[C]: in function 'DataLoader'
	/media/3tb/data/downloads/nix/lab/deepmask/DataLoader.lua:21: in function 'create'
	train.lua:101: in main chunk
	[C]: in function 'dofile'
	/opt/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:145: in main chunk
	[C]: at 0x00406670


my system: gtx 970/ubuntu/4790k/16gb ram

**Comment by tylin** [comment]
https://github.com/facebookresearch/deepmask/issues/14 discusses how to solve this memory issue.

**Comment by ChongqiuMao** [comment]
I encounter the same problem.

**Comment by tongerpeng** [comment]
@hudvin Have you trained your model successfully???
